src/appleseed/renderer/modeling/scene/assembly.cpp

```diff
@@ -56,6 +56,8 @@ namespace renderer
 // Assembly class implementation.
 //
 
+const char* Model = "assembly";
```

Comment:
Although, right now I guess your idea is to stick to a single `Assembly` class that can be populated differently at instantiation-time, right?